src/sentry/eventstore/processing/base.py

```diff
@@ -21,19 +25,23 @@ def __init__(self, inner, timeout=DEFAULT_TIMEOUT):
         self.inner = inner
         self.timeout = timeout
 
-    def _key_for_event(self, event):
-        return cache_key_for_event(event)
-
-    def store(self, event):
-        key = self._key_for_event(event)
+    def store(self, event, unprocessed=False):
+        key = cache_key_for_event(event)
+        if unprocessed:
+            key = _get_unprocessed_key(key)
         self.inner.set(key, event, self.timeout)
         return key
 
-    def get(self, key):
+    def get(self, key, unprocessed=False):
+        if unprocessed:
+            key = _get_unprocessed_key(key)
         return self.inner.get(key)
 
     def delete_by_key(self, key):
-        return self.inner.delete(key)
+        self.inner.delete(key)
+        self.inner.delete("{}:u".format(key))
```

Comment:
1. We can use `_get_unprocessed_key` here
2. This basically means another unconditional call to Redis, right? Any way to restrict it only to specific cases, somehow?

Reply:
If we change this to `delete_many` would it be ok to run it unconditionally?

Reply:
I don't think we can use `delete_many`, because it's not part of the interface now?

Reply:
ah yes inner is not a django cache. I can extend it?

Reply:
it's tricky to do something faster than delete_many, as checking for the feature in here is equally expensive than just attempting to delete